### PR TITLE
Correct some error handling

### LINKED ISSUE #7954: Suspicious error handlings
#### Describe the bug
Some snippets of code returns no error despite checking existence of error.
Reading code around them, I think:
- For 1st, 2nd and 3rd one, probably it's unintentional.
- For 4th one, it can be intentional. But I report it just in case.
https://github.com/grafana/mimir/blob/91dd4380e5c82b883396dee4bb3f485ca924def6/pkg/ruler/remotequerier.go#L223-L226
https://github.com/grafana/mimir/blob/91dd4380e5c82b883396dee4bb3f485ca924def6/pkg/storage/tsdb/block/block_generator.go#L124-L127
https://github.com/grafana/mimir/blob/91dd4380e5c82b883396dee4bb3f485ca924def6/pkg/alertmanager/alertmanager.go#L371-L373
https://github.com/grafana/mimir/blob/91dd4380e5c82b883396dee4bb3f485ca924def6/pkg/ingester/ingester.go#L3228-L3231

#### To Reproduce
N/A

#### Expected behavior
N/A

#### Environment
N/A

#### Additional Context
I'm sorry but I'm reporting it not being familiar with mimir.
I'm developing static checker and scanning public projects to improve the checker. I report detected issues if I consider it's true positive reading codes around them.
Anyway I made effort to get somewhat sure that they are probably unintentional.

### PR DESCRIPTION
**Pull request by seiyab**
#### What this PR does
Correct some error handling reported at https://github.com/grafana/mimir/issues/7954.

#### Which issue(s) this PR fixes or relates to

Fixes #7954 

#### Checklist

- [ ] Tests updated.
- [ ] ~Documentation added.~
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] ~[`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.~

### CODE REVIEW

**Review by grobinson-grafana — APPROVED** [review]

File: pkg/alertmanager/alertmanager.go [hunk 1]
```diff
@@ -369,7 +369,7 @@ func (am *Alertmanager) ApplyConfig(userID string, conf *config.Config, rawCfg s
 		return notifier
 	})
 	if err != nil {
-		return nil
+		return err
```
Comment (grobinson-grafana) [inline pkg/alertmanager/alertmanager.go:372]: `nil` looks like a mistake to me! Thanks for fixing this. LGTM for `alertmanager.go`.

**Review by aknuds1 — APPROVED** [review]
Thanks!

**Comment by seiyab** [comment]
Thank you for super speedy reviews!

**Comment by pstibrany** [comment]
Thank you for your contribution.

**Pull request merged**